### ISSUE TITLE
[DOC] Advertise support for integral types (long, byte etc) tensors

### ISSUE THREAD

**Issue by tvercaut** [f-dangel/unfoldNd#38]
Thank you for this nice extension of unfold. I came here because of the 4d limitation of `torch.nn.functional.unfold` but also because of other limitations including lack of support for integral types, see https://github.com/pytorch/pytorch/issues/44989. And while there is a simple workaround for int64 in `torch.nn.functional.unfold`, this is not the case for `int32`

A quick test seems to indicate that unfoldND doesn't have this integral type limitation. It would be good to highlight this in the documentation and maybe add a unit test for to avoid any regressions?

Here is a quick and dirty example:
```python
!pip install unfoldNd

import torch
print(f'Running PyTorch version: {torch.__version__}')

torchdevice = torch.device('cpu')
if torch.cuda.is_available():
  torchdevice = torch.device('cuda')
  print('Default GPU is ' + torch.cuda.get_device_name(torch.device('cuda')))
print('Running on ' + str(torchdevice))

import unfoldNd as ufnd

im = torch.zeros((1,1,8,16), device=torchdevice, dtype=torch.int64)
im[0,0,3:5,8:10] = 1
print(im.dtype)

plt.figure()
plt.imshow(im.squeeze().cpu().numpy(),cmap="gray")

# Hack to make unfold work with int64 (Note that this won't currently work for int32)
# See https://github.com/pytorch/pytorch/issues/44989#issuecomment-1698832511
ufim = torch.nn.functional.unfold(im.view(torch.float64), kernel_size=3, dilation=4, padding=4, stride=1).view(torch.int64)

plt.figure()
plt.imshow(ufim.squeeze().cpu().numpy(),cmap="gray")

# Now use unfoldND out of the box
# https://github.com/f-dangel/unfoldNd
ufndim = ufnd.unfoldNd(im, kernel_size=3, dilation=4, padding=4, stride=1)

plt.figure()
plt.imshow(ufndim.squeeze().cpu().numpy(),cmap="gray")
```

**Comment by f-dangel** [comment]
Hi Tom,

thanks for your suggestion. I did not know about this limitation in the PyTorch implementation.
Just to be clear I understand it correctly: Say you have some `im` with an unsupported data type, then the workaround is to compute `im_unfold = functional.unfold(im.double()).to(im.dtype)`?

Best,
Felix

**Comment by tvercaut** [comment]
The workaround for `dtype=int64` is to view the data as `float64` instead (this avoids data copy and potential loss of precision):
```pytorch
im_int64_unfold = torch.nn.functional.unfold(im_int64.view(torch.float64)).view(torch.int64)
```
